pkg/agent/utils.go

```diff
@@ -370,6 +378,50 @@ func GetOrderedKubeletConfigFlagString(k map[string]string, cs *datamodel.Contai
 	return buf.String()
 }
 
+func getOrderedKubeletConfigFlagWithCustomConfigurationString(customConfig, defaultConfig map[string]string) string {
+	config := customConfig
+
+	if defaultConfig != nil {
+		for k, v := range defaultConfig {
+			// override configurations from default values to customized ones
+			if _, ok := config[k]; ok {
```

Comment:
nit: you don't need the if statement if you overwrite default config with custom config

on L395 you can range a nil map safely so it doesn't matter if defaultConfig is nil on L384 (assuming this is why you had it flipped to begin with)